### ISSUE TITLE
should consider generic sig and its erasure as overriding for Java member pairs

### ISSUE THREAD

**Issue by adriaanm** [scala/bug#11530]
same / similar / related (?) issue:

```
$> cat Test.java
import java.util.Optional;

public class Test {
  public abstract static class A<T> {
    public void m(Optional<? extends T> a) { }
  }
  public abstract static class B extends A<Object> {
    @Override
    public void m(Optional<?> a) { }
  }
}
$> javac Test.java
$> cat A.scala
class A {
  new Test.B() { }
}
$> qsc A.scala
A.scala:2: error: name clash between inherited members:
def m(x$1: java.util.Optional[_ <: Object]): Unit in class A and
def m(x$1: java.util.Optional[_]): Unit in class B
have same type after erasure: (x$1: java.util.Optional)Unit
  new Test.B() { }
      ^
one error found
```

_Originally posted by @lrytz in https://github.com/scala/bug/issues/11469#issuecomment-488662447_

**Comment by lrytz** [comment]
This is fixed in current 2.13.x, by https://github.com/scala/scala/pull/8049 I assume. I'll submit a test case.

**Comment by lrytz** [comment]
The test is already there in fact. https://github.com/scala/scala/pull/7966/files#diff-0b598992a96dd5e7ec84cd38fc1eeaa0
